src/org/exist/launcher/ServiceManager.java

```diff
@@ -0,0 +1,249 @@
+                launcher.setServiceState();
+            });
+        }
+        return false;
+    }
+
+    boolean start() {
+        if (!isRunning) {
+            runWrapperCmd("start", (code, output) -> {
+                if (code == 0) {
+                    isRunning = true;
+                } else {
+                    LOG.warn("Failed to start service. Exit code: " + code);
+                    JOptionPane.showMessageDialog(null, "Failed to start service. ", "Starting Service Failed", JOptionPane.ERROR_MESSAGE);
+                    isRunning = false;
+                }
+                launcher.setServiceState();
+            });
+        }
+        return isRunning;
+    }
+
+    boolean stop() {
+        if (isRunning) {
+            runWrapperCmd("stop", (code, output) -> {
+                if (code == 0) {
+                    isRunning = false;
+                } else {
+                    LOG.warn("Failed to stop service. Exit code: " + code);
+                    JOptionPane.showMessageDialog(null, "Failed to stop service.", "Stopping Service Failed", JOptionPane.ERROR_MESSAGE);
+                    isRunning = true;
+                }
+                launcher.setServiceState();
+            });
+        }
+        return !isRunning;
+    }
+
+    private String getShellCmd(String cmd) {
+        if (SystemUtils.IS_OS_UNIX) {
+            return cmd + "Daemon.sh";
+        }
+        return cmd + "Service.bat";
+    }
+
+    private void isRoot(Consumer<Boolean> consumer) {
+        final List<String> args = new ArrayList<>(2);
+        args.add("id");
+        args.add("-u");
+        run(args, (code, output) -> {
+            consumer.accept("0".equals(output.trim()));
+        });
+    }
+
+    void showServicesConsole() {
+        final List<String> args = new ArrayList<>(2);
+        args.add("cmd");
+        args.add("/c");
+        args.add("services.msc");
+
+        run(args, null);
+    }
+
+    private void runWrapperCmd(final String cmd, final BiConsumer<Integer, String> consumer) {
+        final Path executable = wrapperDir.resolve("bin/" + getShellCmd(cmd));
+        final List<String> args = new ArrayList<>(1);
+        args.add(executable.toString());
+
+        run(args, (code, output) -> {
+            if (LOG.isDebugEnabled()) {
+                LOG.debug(output);
+            }
+            consumer.accept(code, output);
+        });
+    }
+
+    public static void run(List<String> args, BiConsumer<Integer, String> consumer) {
+        final ProcessBuilder pb = new ProcessBuilder(args);
+        final Optional<Path> home = ConfigurationHelper.getExistHome();
+
+        pb.directory(home.orElse(Paths.get(".")).toFile());
+        pb.redirectErrorStream(true);
+        if (consumer == null) {
+            pb.inheritIO();
+        }
+        try {
+            final Process process = pb.start();
+            if (consumer != null) {
+                final StringBuilder output = new StringBuilder();
```

Comment:
It would seem to me that Java's NIO.2 [`Files.readAllLines`](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#readAllLines%28java.nio.file.Path,%20java.nio.charset.Charset%29) or even [`Files.readAllBytes`](https://docs.oracle.com/javase/7/docs/api/java/nio/file/Files.html#readAllBytes%28java.nio.file.Path%29) would be a better choice here.